tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -179,4 +205,19 @@ object ScatterGather {
       throw new UnsupportedOperationException("getTasks is not supported and should never be called on sub-scatters.")
     }
   }
+
+  /**
+    * A type of partitioner that is found at stage 2 and beyond where the input is guaranteed to be a task already, and
+    * not part of some domain type.  The given method `f` is used to group the source tasks.
+    */
+  private class GroupByPartitioner[Result <: Task, Key](f: Result => Key) extends SimpleInJvmTask with Partitioner[(Key, Seq[Result])] {
+    var _partitions: Option[Seq[(Key, Seq[Result])]] = None
```

Comment:
Why the underscore?  Can you not just drop it and add in `override` and drop the method below?

Reply:
I think I was worried that `partitions` was a public method, and I didn't want it to be set outside this scope, but I omitted making it private.  I don't think it matters since `GroupByPartitioner` is private to this scope anyhow.